nameko/rpc.py

```diff
@@ -387,19 +422,22 @@ def _call(self, *args, **kwargs):
         # asynchronously and conditionally, so we can't wait() on the channel
         # for it (will wait forever on successful delivery).
         #
-        # Instead, we use (the rabbitmq extension) confirm_publish
+        # Instead, we make use of (the rabbitmq extension) confirm_publish
         # (https://www.rabbitmq.com/confirms.html), which _always_ sends a
         # reply down the channel. Moreover, in the case where no queues are
         # bound to the exchange (service unknown), the basic.return is sent
         # first, so by the time kombu returns (after waiting for the confim)
         # we can reliably check for returned messages.
 
+        # Note that overriding :attr:`self.use_confirms` will disable this
+        # functionality and therefore :class:`UnknownService` will never
+        # be raised (and the caller will hang).
+
         routing_key = '{}.{}'.format(self.service_name, self.method_name)
 
         exchange = get_rpc_exchange(container.config)
 
-        with producers[conn].acquire(block=True) as producer:
-
+        def publish(producer):
```

Comment:
Actually not sure. I've unrolled it so it's more inline with the other producers